### DISCUSSION TITLE
Notifications / Inbox

### DISCUSSION THREAD

**Discussion by JordanKnott** [JordanKnott/taskcafe#57] [Ideas]
Some example notifications are

- Someone marks a task that I am a member of complete
- Someone changes the title of a task that I am a member of
- Someone adds me to a task
- Someone adds me to a project

If similar tasks occur within 5 minutes of each other, they get merged or reversed. E.g

If I add someone to a task, then remove them within 5 minutes of adding them - no notification is sent

If I add someone to 10 tasks within 5 minutes of each other, then only one group notification would be sent out instead of 10 individual ones

The number of unread notifications are shown in a badge on the bell in the top right navbar.
Clicking the navbar opens a popup that shows the most recent notifications (last six?). It also has a link to the inbox that shows all the unread notifications (with option to show read ones as well).

**Alternatively, the bell could just take you to the inbox instead?**

The inbox will also have a tab on project activity, which will be an upcoming feature.

I'm open to suggestions and thoughts.

**Comment by obbardc** [comment]
How about notification when task due date is near, maybe per task - one day before, one hour before etc ?

How about email notifications too- realtime and daily digest?

**Comment by JordanKnott** [comment]
Due date reminders will be a feature in the future as well as daily (and weekly digests) - just gotta decide on how I want to implement it
